homeassistant/components/androidtv/media_player.py

```diff
@@ -0,0 +1,336 @@
+                          if dev.entity_id in entity_id]
+
+        for target_device in target_devices:
+            output = target_device.adb_command(cmd)
+
+            # log the output if there is any
+            if output:
+                _LOGGER.info("Output of command '%s' from '%s': %s",
+                             cmd, target_device.entity_id, repr(output))
+
+    hass.services.register(ANDROIDTV_DOMAIN, SERVICE_ADB_COMMAND,
+                           service_adb_command,
+                           schema=SERVICE_ADB_COMMAND_SCHEMA)
+
+
+def adb_decorator(override_available=False):
+    """Send an ADB command if the device is available and catch exceptions."""
+    def _adb_decorator(func):
+        """Wait if previous ADB commands haven't finished."""
+        @functools.wraps(func)
+        def _adb_exception_catcher(self, *args, **kwargs):
+            # If the device is unavailable, don't do anything
+            if not self.available and not override_available:
+                return None
+
+            try:
+                return func(self, *args, **kwargs)
+            except self.exceptions as err:
+                _LOGGER.error(
+                    "Failed to execute an ADB command. ADB connection re-"
+                    "establishing attempt in the next update. Error: %s", err)
+                self._available = False  # pylint: disable=protected-access
+                return None
+
+        return _adb_exception_catcher
+
+    return _adb_decorator
+
+
+class AndroidTVDevice(MediaPlayerDevice):
+    """Representation of an Android TV device on the network."""
+
+    def __init__(self, atv, name, apps):
+        """Initialize the Android TV device."""
+        from androidtv import ACTIONS
+        self.actions = ACTIONS
+        self.apps = apps
+
+        self.androidtv = atv
+
+        self._name = name
+
+        # ADB exceptions to catch
+        if not self.androidtv.adb_server_ip:
+            # Using "python-adb" (Python ADB implementation)
+            from adb.adb_protocol import (InvalidChecksumError,
+                                          InvalidCommandError,
+                                          InvalidResponseError)
+            from adb.usb_exceptions import TcpTimeoutException
+
+            self.exceptions = (AttributeError, BrokenPipeError, TypeError,
+                               ValueError, InvalidChecksumError,
+                               InvalidCommandError, InvalidResponseError,
+                               TcpTimeoutException)
+        else:
+            # Using "pure-python-adb" (communicate with ADB server)
+            self.exceptions = (ConnectionResetError,)
+
+        self._state = None
+        self._available = self.androidtv.available
+        self._current_app = None
+        self._device = None
+        self._muted = None
+        self._properties = self.androidtv.properties
+        self._unique_id = 'androidtv-{}-{}'.format(
```

Comment:
Looks like we are getting a `unique_id` but we are not doing anything with it yet.  Probably want to return this for the entity registry to work as expected.

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/21674/commits/4fbdd538cabdea0720e4c0c5c512c1651019a7c1